js/views/migration_view.js

```diff
@@ -0,0 +1,179 @@
+;(function () {
+  'use strict';
+  window.Whisper = window.Whisper || {};
+
+  var State = {
+    DISCONNECTING: 1,
+    EXPORTING: 2,
+    COMPLETE: 3
+  };
+
+  Whisper.Migration = {
+    isComplete: function() {
+      return storage.get('migrationState') === State.COMPLETE;
+    },
+    inProgress: function() {
+      return storage.get('migrationState') > 0 || this.everComplete();
+    },
+    markComplete: function(target) {
+      storage.put('migrationState', State.COMPLETE);
+      storage.put('migrationEverCompleted', true);
+      if (target) {
```

Comment:
Should definitely be, yes. But if it's `undefined`, like it happened to be during my testing yesterday, that `put()` call blows up. :0/